src/AutoMapper/Execution/ReflectionHelper.cs

```diff
@@ -136,47 +136,78 @@ public static MemberInfo FindProperty(LambdaExpression lambdaExpression)
-                return ((MethodInfo) memberInfo).ReturnType;
+                return ((MethodInfo)memberInfo).ReturnType;
             if (memberInfo is PropertyInfo)
-                return ((PropertyInfo) memberInfo).PropertyType;
+                return ((PropertyInfo)memberInfo).PropertyType;
             if (memberInfo is FieldInfo)
-                return ((FieldInfo) memberInfo).FieldType;
+                return ((FieldInfo)memberInfo).FieldType;
             return null;
         }
-        
-		public static IMemberGetter ToMemberGetter(this MemberInfo accessorCandidate)
-		{
-			if (accessorCandidate == null)
-				return null;
-
-			if (accessorCandidate is PropertyInfo)
-                return new PropertyGetter((PropertyInfo) accessorCandidate);
-
-			if (accessorCandidate is FieldInfo)
-                return new FieldGetter((FieldInfo) accessorCandidate);
-
-			if (accessorCandidate is MethodInfo)
-                return new MethodGetter((MethodInfo) accessorCandidate);
-
-			return null;
-		}
-        
-		public static IMemberAccessor ToMemberAccessor(this MemberInfo accessorCandidate)
-		{
-			var fieldInfo = accessorCandidate as FieldInfo;
-			if (fieldInfo != null)
-                return accessorCandidate.DeclaringType.IsValueType()
-                    ? (IMemberAccessor) new ValueTypeFieldAccessor(fieldInfo)
-                    : new FieldAccessor(fieldInfo);
-
-			var propertyInfo = accessorCandidate as PropertyInfo;
-			if (propertyInfo != null)
-                return accessorCandidate.DeclaringType.IsValueType()
-                    ? (IMemberAccessor) new ValueTypePropertyAccessor(propertyInfo)
-                    : new PropertyAccessor(propertyInfo);
-
-			return null;
-		}
+
+        public static IMemberGetter ToMemberGetter(this MemberInfo accessorCandidate)
+        {
+            try
+            {
+                if (accessorCandidate == null)
+                    return null;
+
+                if (accessorCandidate is PropertyInfo)
+                    return
+                        Activator.CreateInstance(
+                            typeof (PropertyAccessor<,>).MakeGenericType(accessorCandidate.DeclaringType,
+                                accessorCandidate.GetMemberType()), accessorCandidate) as IMemberGetter;
+
+                if (accessorCandidate is FieldInfo)
+                    return
+                        Activator.CreateInstance(
+                            typeof (FieldGetter<,>).MakeGenericType(accessorCandidate.DeclaringType,
+                                accessorCandidate.GetMemberType()), accessorCandidate) as IMemberGetter;
+
+                if (accessorCandidate is MethodInfo)
+                    return
+                        Activator.CreateInstance(
+                            typeof (MethodGetter<,>).MakeGenericType(accessorCandidate.DeclaringType,
+                                accessorCandidate.GetMemberType()), accessorCandidate) as IMemberGetter;
+
```

Comment:
They differ on typeof(), but everything else is the same yes.  You could make a function and just pass in the type and save a few lines.  I left it this way because I didn't want to differentiate too much from the original.

I'm not opposed to changing it in the future though.